src/main/java/uk/gov/register/util/HashValue.java

```diff
@@ -0,0 +1,65 @@
+package uk.gov.register.util;
+
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.fasterxml.jackson.annotation.JsonValue;
+import uk.gov.register.core.HashingAlgorithm;
+import uk.gov.register.exceptions.HashDecodeException;
+
+public class HashValue {
+    private final String value;
+    private final String hashingAlgorithm;
+
+    public HashValue(HashingAlgorithm hashingAlgorithm, String value) {
+        this.hashingAlgorithm = hashingAlgorithm.toString();
+        this.value = value;
+    }
+
+    public HashValue(String hashingAlgorithm, String value) {
+        this.hashingAlgorithm = hashingAlgorithm;
+        this.value = value;
+    }
+
+    @JsonValue
```

Comment:
`@JsonValue` should not be required here. You have used `@JsonSerialize(using = ToStringSerializer.class)` in the relevant places so it is using the `toString()` method anyway (which points to `encode()`).